presto-orc/src/test/java/com/facebook/presto/orc/LongDictionaryProviderTest.java

```diff
@@ -0,0 +1,291 @@
+import com.facebook.presto.orc.stream.LongOutputStreamDwrf;
+import com.facebook.presto.orc.stream.OrcInputStream;
+import com.facebook.presto.orc.stream.SharedBuffer;
+import com.facebook.presto.orc.stream.StreamDataOutput;
+import com.facebook.presto.orc.stream.ValueInputStream;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.Slice;
+import io.airlift.units.DataSize;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.orc.OrcDecompressor.createOrcDecompressor;
+import static com.facebook.presto.orc.checkpoint.Checkpoints.getDictionaryStreamCheckpoint;
+import static com.facebook.presto.orc.metadata.CompressionKind.SNAPPY;
+import static com.facebook.presto.orc.metadata.OrcType.OrcTypeKind.LONG;
+import static com.facebook.presto.orc.metadata.Stream.StreamKind.DICTIONARY_DATA;
+import static com.facebook.presto.orc.stream.CheckpointInputStreamSource.createCheckpointStreamSource;
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.airlift.units.DataSize.Unit.KILOBYTE;
+import static java.lang.Math.toIntExact;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotEquals;
+
+public class LongDictionaryProviderTest
+{
+    private static final OrcDataSourceId ORC_DATA_SOURCE_ID = new OrcDataSourceId("dict_provider_test");
+    private static final DataSize COMPRESSION_BLOCK_SIZE = new DataSize(256, KILOBYTE);
+    private static final OrcType LONG_TYPE = new OrcType(LONG, ImmutableList.of(), ImmutableList.of(), Optional.empty(), Optional.empty(), Optional.empty());
+    private static final DummyOrcDataSource DUMMY_ORC_DATA_SOURCE = new DummyOrcDataSource();
+
+    @Test(dataProvider = "dataForDictionaryLoadingTest")
+    public void testLongDictionaryLoading(Map<NodeId, long[]> dictionaryStreams, List<NodeId> unused)
+            throws Exception
+    {
+        LongDictionaryProvider longDictionaryProvider = new LongDictionaryProvider(createLongDictionaryStreamSources(dictionaryStreams));
+        for (Map.Entry<NodeId, long[]> entry : dictionaryStreams.entrySet()) {
+            StreamId streamId = entry.getKey().toDictionaryDataStreamId();
+            long[] data = entry.getValue();
+            long[] dictionary = new long[0];
+            long[] newDictionary = longDictionaryProvider.getDictionary(createFlatStreamDescriptor(streamId), dictionary, data.length);
+            assertNotEquals(newDictionary, dictionary);
+            assertEquals(newDictionary, data);
+        }
+    }
+
+    @Test(dataProvider = "dataForTestDataCorruptionCheck", expectedExceptions = OrcCorruptionException.class, expectedExceptionsMessageRegExp = ".* Dictionary is not empty but data stream is not present.*")
```

Comment:
Replace `expectedExceptions` with assertThrows/expectThrows in the method body. It allows to make more precise assertions.

Reply:
This currently works well enough, so I'm not inclined to change it. I used to work a little bit in FbEtl's java code base, and they strongly prefer to stick to only testng framework. I didn't fine these options in testng framework, thus the current setup. Please let me know if you feel very strongly against it.